daft/udf.py

```diff
@@ -170,47 +169,18 @@ def get_args_for_slice(start: int, end: int):
 
 
 @dataclasses.dataclass
-class UDF:
+class CommonUDFArgs:
```

Comment:
is `CommonUDFArgs` supposed to ever be accessed by a user (either getting or setting?) If so, it feels kind of unergonomic to have to do `udf.common_args.resource_request`. If not, could we prefix the instance variables in `StatefulUDF` and `StatelessUDF` with an underscore?

Reply:
I think we're not going to expose it to users. Yes, we can add an underscore!

Reply:
Ah... I did this because the dataclasses don't support dunder for attributes I believe. Will leave it as such.